yew-functional/src/lib.rs

```diff
@@ -308,6 +522,8 @@ where
     );
 }
 
+/// Sometimes, it's needed to manually define dependencies for [`use_effect`].
+/// In such cases, we use `use_effect_with_deps`.
```

Comment:
Let's expand this a bit
```suggestion
/// This hook is similar to [`use_effect`] but it accepts dependencies.
///
/// Whenever the dependencies are changed, the effect callback is called again.
/// To detect changes, dependencies must implement `PartialEq`.
/// Note that the destructor also runs when dependencies change.
```

Not great but better than the current one